spec/property_sets_spec.rb

```diff
@@ -263,6 +263,8 @@ class AnotherThing < ActiveRecord::Base
         { :id => account.texts.lookup(:bar).id, :name => "bar", :value => "1" }
       ])
 
+      account.reload
+
       expect(account.texts.foo).to eq("0")
       expect(account.texts.bar).to eq("1")
     end
```

Comment:
These were the only changes needed in the test files. I couldn't find a way around adding `#to_a`. I'm not a big fan of how an AR:Relation can automatically be reified behave like an Enumerable when required, so I didn't try to implement it in the new `AssociationWrapper`.

On the `#reload`, this is necessary because this test uses `account.update_attributes!(:texts_attributes => [...])`, which bypasses the `property_set` gem and goes directly to ActiveRecord. I couldn't find a convenient hook to invalidate the cache automatically.

Compare this to the next test below (`it "be updateable as a nested structure"`), that instead  uses `account.update_attributes!(:settings => { ... })`. That _is_ intercepted by the `property_set` gem, and technically there the `#reload` is not necessary (even though the test was calling it 🤷 ).